qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,240 @@
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.AppointmentBlocksPage;
+import org.openmrs.contrib.qaframework.page.AppointmentSchedulingPage;
+import org.openmrs.contrib.qaframework.page.ManageAppointmentsPage;
+import org.openmrs.contrib.qaframework.page.ManageProviderSchedulesPage;
+import org.openmrs.contrib.qaframework.page.RequestAppointmentPage;
+
+public class AppointmentSteps extends Steps {
+
+	private AppointmentSchedulingPage appointmentSchedulingPage;
+	private AppointmentBlocksPage appointmentBlocksPage;
+	private ManageAppointmentsPage manageAppointmentsPage;
+	private ManageProviderSchedulesPage manageProviderSchedulesPage;
+	private RequestAppointmentPage requestAppointmentPage;
+	private TestData.PatientInfo testPatient;
+	private static final String FIRST_SERVICE_NAME = "Oncology";
+	private static final String SECOND_SERVICE_NAME = "Dermatology";
+
+	@Before(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void visitDashboard() {
+		initiateWithLogin();
+		testPatient = createTestPatient();
+		findPatientPage = homePage.goToFindPatientRecord();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_APPOINTMENT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Request Appointment link from the Patient dashboard")
+	public void loadRequestAppointmentPage() {
+		requestAppointmentPage = (RequestAppointmentPage) dashboardPage.clickOnRequest();
+	}
+
+	@When("the system loads Request Appointment page")
+	public void systemLoadsRequestAppointmentPage() {
+		assertTrue(textExists("Request Appointment"));
+	}
+
+	@And("a user fills in the appointment request details")
+	public void captureAppointmentRequestDetails() {
+		requestAppointmentPage.enterAppointmentType("Oncology");
+		requestAppointmentPage.enterMinimumValue("0");
+		requestAppointmentPage.selectMinimumUnits("Day(s)");
+		requestAppointmentPage.enterMaximumValue("2");
+		requestAppointmentPage.selectMaximumUnits("Day(s)");
+		requestAppointmentPage.addAppointmentRequestNote("I need to see a doctor first");
+	}
+
+	@And("a user clicks save button")
+	public void saveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
```

Comment:
How does the above prove that the appointment request has been added to the table?

Reply:
@dkayiwa  this proves that the appointment request list is not empty after adding the request since the test assumes that it is the only request that has been saved.